java/statestore/src/main/java/sleeper/statestore/dynamodb/DynamoDBFileReferenceStore.java

```diff
@@ -127,6 +128,56 @@ public void addFiles(List<FileReference> fileReferences) throws StateStoreExcept
         }
     }
 
+    @Override
+    public void splitFileReferences(List<SplitFileReferenceRequest> splitRequests) throws StateStoreException {
+        long updateTime = clock.millis();
+        for (SplitFileReferenceRequest splitRequest : splitRequests) {
```

Comment:
Given that calling splitFileReference will fail if there are more than 99 requests in splitRequests I suggest we validate the entire list of SplitFileReferenceRequests before attempting to execute them. The validation rule would be that the size of splitRequests is <= 99. If that is not true for any request in the list, then we should fail early with an exception rather than completing some of them and then failing.

Reply:
Should we avoid making assumptions about the state store implementation before calling in to the state store? The issue with transaction size is only a problem for DynamoDB, so maybe we should resolve that inside the DynamoDB implementation?

Reply:
Not sure. Either:

- The method accepts a list of requests which it can't satisfy in one transaction (in which case it might fail halfway through the list and the request will be half-satisfied and the caller will get an exception - if the exception tells them that say the first 10 items in a list succeeded but the 11th failed that would be ok as the client could in principle just retry the failed ones)
- Or the method validates whether it can do everything in the list in one go, in which case the client needs to know which state store it's talking to and create appropriately sized lists.

Probably option 1 is best, with the exception reporting the progress through the list.